File: PruebaTecnicaApp/ClientApp/src/app/administrador/modules/producto/class/producto.ts
import { TipoProducto } from '../../tipo-producto/class/tipo';

export class Producto {
    id: number = 0;
    idTipoProducto: number = 0;
    nombre?: string;
    descripcion?: string;
    existencia: number = 0;
    valor: number = 0;

    tipoProducto?: TipoProducto;
}

File: Entidad/Producto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Entidad
{
    public class Producto
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("TipoProducto")]
        public int IdTipoProducto { get; set; }
        public TipoProducto? TipoProducto { get; set; }
        public string? Nombre { get; set; }
        public string? Descripcion { get; set; }
        public double Existencia { get; set; }

        [DataType(DataType.Currency)]
        [Column(TypeName = "money")]
        public decimal Valor { get; set; }

    }
}